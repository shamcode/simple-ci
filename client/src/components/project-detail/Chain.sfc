<template>
    {% import path from 'sham-ui-router/lib/params' %}

    <button
        class="project-chain"
        :onclick={{this$._toggleExpand}}
    >
        {{chain.name}}
    </button>

    {% if isExpand %}
        <dl class="project-chain-actions">
            <dt>
                <a :hrefto={{path("project-chain-run").param("id", project.id).param("chainId", chain.id)}}>
                    Run
                </a>
            </dt>
            <dd>Execute chain</dd>
            <dt>
                <a :hrefto={{path("project-chain-detail").param("id", project.id).param("chainId", chain.id)}}>
                    Edit
                </a>
            </dt>
            <dd>Edit chain name & command</dd>
            <dt>
                <a
                    :hrefto={{path("project-chain-delete").param("id", project.id).param("chainId", chain.id)}}
                    class="chain-delete"
                >
                    Delete
                </a>
            </dt>
            <dd>Delete chain</dd>
        </dl>
    {% endif %}
</template>

<script>

    function Chain( options ) {
        const isExpand = $();

        const state = options( {
            [ isExpand ]: false,
            [ $.chain ]: {},
            [ $.project ]: { chains: [] }
        } );

        this$._toggleExpand = e => {
            e.preventDefault();
            state[ isExpand ] = !state[ isExpand ];
        };
    }

    export default Component( Template, Chain );
</script>
