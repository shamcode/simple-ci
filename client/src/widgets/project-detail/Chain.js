import { options } from 'sham-ui';
import ChainTemplate from './Chain.sht';

export default class Chain extends ChainTemplate {
    @options get chain() {
        return {};
    }
    @options isExpand = false;
    @options get project() {
        return { chains: [] };
    }

    _routerParams( project, chain ) {
        return {
            id: project === undefined ? null : project.id,
            chainId: chain === undefined ? null : chain.id
        };
    }

    _toggleExpand( e ) {
        e.preventDefault();
        this.update( {
            isExpand: !this.options.isExpand
        } );
    }
}
