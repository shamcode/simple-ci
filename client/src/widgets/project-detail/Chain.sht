{% import LinkTo from 'sham-ui-router/link-to' %}
<button
        class="project-chain"
        :onclick={{this._toggleExpand.bind(this)}}
>
    {{chain.name}}
</button>

{% if isExpand %}
    <dl class="project-chain-actions">
        <dt>
            <a href="#">
                Edit
            </a>
        </dt>
        <dd>Edit chain name & steps</dd>
        <dt>
            <LinkTo
                    text="Delete"
                    className="chain-delete"
                    path="project-chain-delete"
                    params={{this._routerParams( project, chain )}}
            />
        </dt>
        <dd>Delete chain</dd>
    </dl>
{% endif %}