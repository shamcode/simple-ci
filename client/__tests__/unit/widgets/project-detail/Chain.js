import Chain from '../../../../src/widgets/project-detail/Chain';
import * as directives from 'sham-ui-directives';
import renderer from 'sham-ui-test-helpers';

it( 'renders correctly', () => {
    const meta = renderer( Chain, {
        directives
    } );
    expect( meta.toJSON() ).toMatchSnapshot();
} );
