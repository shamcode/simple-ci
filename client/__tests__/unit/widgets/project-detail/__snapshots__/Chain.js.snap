// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Object {
  "ID": "widget",
  "constructorOptions": Object {
    "directives": Object {
      "onclick": [Function],
      "onsubmit": [Function],
    },
  },
  "html": "
<button class=\\"project-chain\\">null</button>
<!--if-->
",
  "nodes": Array [
    Object {
      "ID": undefined,
      "constructorOptions": null,
      "html": null,
      "nodes": null,
      "options": null,
    },
    Object {
      "ID": undefined,
      "constructorOptions": null,
      "html": null,
      "nodes": null,
      "options": null,
    },
  ],
  "options": Object {
    "actionSequence": Array [
      "render",
      "bindEvents",
    ],
    "chain": Object {},
    "directives": Object {
      "onclick": [Function],
      "onsubmit": [Function],
    },
    "isExpand": false,
    "project": Object {
      "chains": Array [],
    },
    "renderAsync": false,
    "renderDependence": Array [],
    "types": Array [],
  },
}
`;
